sympy/matrices/expressions/matexpr.py

```diff
@@ -187,9 +186,12 @@ def as_real_imag(self, deep=True, **hints):
         return (real, im)
 
     def _eval_inverse(self):
-        from sympy.matrices.expressions.inverse import Inverse
         return Inverse(self)
 
+    def _eval_determinant(self):
+        from sympy.matrices.expressions.determinant import Determinant
+        return Determinant(self)
```

Comment:
Can the import go at the bottom of the file?

Reply:
It should indeed be possible (as well as some other corresponding ones). I was a bit unsure why they were at the bottom of the file and try to follow (one of) the style of the other methods.

Reply:
The imports are at the bottom to avoid circularity. Here MatrixExpr is a superclass but needs to import its own subclasses. Adding imports in functions like this is a quick fix but actually has a runtime cost. Adding imports at the bottom of the file is potentially trickier because you sometimes need to figure out exactly where in the sequence of imports it can go and which go at the top of the file and which at the bottom etc.

Personally I think this shows one reason why methods like `det` are a bad idea. It would be better to define the class in one file and then functions that operate on the class in another with no need for the class to import the downstream functions.